Flaky e2e test (cron-fallback)

We need to fix this flaky e2e test:

```
--- FAIL: TestRunCronExample (310.61s)
    --- PASS: TestRunCronExample/cron (127.61s)
    --- PASS: TestRunCronExample/cron-timer (60.37s)
    --- FAIL: TestRunCronExample/cron-fallback (91.99s)
```

It happens often, e.g. https://github.com/apache/camel-k/runs/1747588145?check_suite_focus=true, but disappears after running the CI multiple times.

--- a/pkg/util/camel/camel_runtime_catalog.go
+++ b/pkg/util/camel/camel_runtime_catalog.go
@@ -36,6 +36,15 @@ func NewRuntimeCatalog(spec v1.CamelCatalogSpec) *RuntimeCatalog {
 	for id, artifact := range catalog.Artifacts {
 		for _, scheme := range artifact.Schemes {
 			scheme := scheme
+
+			// In case of duplicate only, choose the "org.apache.camel.quarkus" artifact (if present).
+			// Workaround for https://github.com/apache/camel-k-runtime/issues/592
+			if _, duplicate := catalog.artifactByScheme[scheme.ID]; duplicate {
+				if artifact.GroupID != "org.apache.camel.quarkus" {
+					continue
+				}
+			}
+
 			catalog.artifactByScheme[scheme.ID] = id
 			catalog.schemesByID[scheme.ID] = scheme
 		}

